test/client/snapshot-request.js

```diff
@@ -0,0 +1,430 @@
+var Backend = require('../../lib/backend');
+var expect = require('expect.js');
+var lolex = require("lolex");
+var types = require('../../lib/types');
+
+describe('SnapshotRequest', function () {
+  var backend;
+  var clock;
+  var json0 = types.map['json0'];
+
+  var DAY0 = new Date("2018-05-30");
+  var DAY1 = new Date("2018-06-01");
+  var DAY2 = new Date("2018-06-02");
+  var DAY3 = new Date("2018-06-03");
+  var DAY4 = new Date("2018-06-04");
+  var ONE_DAY = 1000 * 60 * 60 * 24;
+
+  beforeEach(function () {
+    clock = lolex.install({
+      now: DAY1,
+    });
+
+    backend = new Backend();
+  });
+
+  afterEach(function (done) {
+    clock.uninstall();
+
+    backend.close(done);
+  });
+
+  describe('a document with some simple versions a day apart', function () {
+    var emptySnapshot = {
```

Comment:
It would be nice to update the variable names for clarity:

- emptySnapshot -> v0
- v0 -> v1
- v1 -> v2
- v2 -> v3